---
title: "Pan-Collagen Gene Survival Analysis in Ovarian Cancer"
author: "Robert Hodges"
date: "4/22/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
df <- read.csv(file = "D:\\Data Science\\Thesis\\CleanedOVData.csv")
```


```{r}
### DV is time to event and IV are the stratified gene columns.

### Primary Copy Number Variations being looked at in 568 patients are...

### -less than 50 fold expression -  COL1A1, COL5A2, COL12A1 and COL17A1

### -greater than 50 fold expression - COL1A2, COL15A1 and COL21A1

### -extremely high level of expression - COL3A1

### -also look at COL22A1
```

```{r}
###  Numbers in dataframe for each gene reference the number of gene copies.  Sometimes we have gene deletions and sometimes extra copies.  -2 = complete deletion, -1 = partial deletion, 0 = normal, 1 = partial duplication, 2 = complete duplication
```

```{r}
library(ggplot2)
library(survival)
library(survminer)
library(dplyr)
library(tidyverse)
library(funModeling) 
library(Hmisc)
library(DataExplorer)
library(splines)
library(lattice)
```

```{r}
### EDA
head(df)
# install.packages("funModeling")
# install.packages("Hmisc")
#install.packages("DataExplorer") 
```

```{r}
### Removing junk columns
df$LRCOL1 <- NULL   #This isn't a collagen gene
df$MCOLN1 <- NULL   #This isn't a collagen gene
df$MCOLN2 <- NULL   #This isn't a collagen gene
df$MCOLN3 <- NULL   #This isn't a collagen gene
```

```{r}
summary(df)
glimpse(df)
plot_missing(df)  #Data explorer package.  Shows no missing values in dataset.  Good.
plot_histogram(df)  #Shows histogram plots of all genes in the dataframe.  This is interesting to look at.  Note that these are collagen gene CNV for the entire dataframe of ovarian tumors, not just collagen's most commonly expressed genes in ovarian tumors which were listed at the beginning of the document.  So simply put, all these genes are found in ovarian tumors, but some are much more expressed than others.  However, please realize that gene expression and CNV are not the same thing and are not correlated.  Therefore, just because a gene has high expression doesn't mean it's CNV will affect anything.  But, this is good to illustrate that a gene like COL4A3BP has lots of partial CNV deletions compared to the gene COL9A3 which is mostly normal CNV or partial duplications.  There is also the plot_density(df) function which shows the same thing but in a density chart.
plot_bar(df$Race)  ## Looking at race column in dataframe.  Mostly causacian females (85%).
plot_histogram(df$Days.to.Death)
```

```{r}
#####################
### DATA CLEANING ###
#####################

### Most cleaning done in Alteryx as it was easier for me and the data cleaning was quite complex.  R was giving me a lot of trouble.

#describe(df$Days.to.Death)  #Need to find missing value
#is.na(df$Days.to.Death)
### Found it in row 240.  Went back to original file and checked the values from TCGA database.
```

```{r}
#################
###  COL1A1  ####
#################

### Description of gene column numbers

#df$COL1A1[df$COL1A1 == -2] <- 'Complete Deletion'
#df$COL1A1[df$COL1A1 <= -1] <- 'Partial Deletion'
#df$COL1A1[df$COL1A1 == 0] <- 'Normal'
#df$COL1A1[df$COL1A1 >= 1] <- 'Partial Duplication'
#df$COL1A1[df$COL1A1 == 2] <- 'Complete Duplication'

#Splicing these into 3 groups.

df[df$COL1A1 == -2,]$COL1A1 <- 'Deletion'
df[df$COL1A1 == -1,]$COL1A1 <- 'Deletion'
df[df$COL1A1 == 0,]$COL1A1 <- 'Normal'
df[df$COL1A1 == 1,]$COL1A1 <- 'Duplication'
df[df$COL1A1 == 2,]$COL1A1 <- 'Duplication'
```

```{r}
### Briefly looking at race
survRace <- survfit(Surv(Days.to.Death, status) ~ Race, data = df)
print(survRace)

ggsurvplot(survRace,
         pval = TRUE, conf.int = F,
         risk.table = TRUE,
         risk.table.col = "strata",
         linetype = "strata",
         surv.median.line = "hv",
         ggtheme = theme_bw())

logrankRace <- survdiff(Surv(Days.to.Death, status) ~ Race, data = df)
print(logrankRace)
### While this is statistically significant, the data seems incomplete and biased.  I would like to see more data for other races in this for comparison.  It may be possible that there is actually a survival difference between races, which may be due to multiple issues with regards to genetics, healthcare, income, education, and other factors.  However, better data is needed before we can start to make inferences here.
```

```{r}
#df$COL1A1 <- factor(df$COL1A1) #Factored gene column

df$Days.to.Death <- as.numeric(as.character(df$Days.to.Death))  #Have to cast column to character first and then to numeric as it is factored.

survCOL1A1 <- survfit(Surv(Days.to.Death, status) ~ COL1A1, data = df)
print(survCOL1A1)

ggsurvplot(survCOL1A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL1A1 <- survdiff(Surv(Days.to.Death, status) ~ COL1A1, data = df)
print(logrankCOL1A1)
### No findings.
```

```{r}
#################
###  COL5A2  ####
#################

df[df$COL5A2 == -2,]$COL5A2 <- 'Deletion'
df[df$COL5A2 <= -1,]$COL5A2 <- 'Deletion'
df[df$COL5A2 == 0,]$COL5A2 <- 'Normal'
df[df$COL5A2 == 1,]$COL5A2 <- 'Duplication'
df[df$COL5A2 == 2,]$COL5A2 <- 'Duplication'

survCOL5A2 <- survfit(Surv(Days.to.Death, status) ~ COL5A2, data = df)
print(survCOL5A2)

ggsurvplot(survCOL5A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL5A2 <- survdiff(Surv(Days.to.Death, status) ~ COL5A2, data = df)
print(logrankCOL5A2)
### No Findings.
```

```{r}
##################
###  COL12A1  ####
##################

#df[df$COL12A1 == -2,]$COL12A1 <- 'Deletion'   #This was deleted and added into the next line of coding with a <= as there wasn't any or enough rows for this code, thus, errors.
df[df$COL12A1 == 0,]$COL12A1 <- 'Normal'
df[df$COL12A1 <= -1,]$COL12A1 <- 'Deletion'
df[df$COL12A1 == 1,]$COL12A1 <- 'Duplication'
df[df$COL12A1 == 2,]$COL12A1 <- 'Duplication'

survCOL12A1 <- survfit(Surv(Days.to.Death, status) ~ COL12A1, data = df)
print(survCOL12A1)

ggsurvplot(survCOL12A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL12A1 <- survdiff(Surv(Days.to.Death, status) ~ COL12A1, data = df)
print(logrankCOL12A1)
### Found something!  I will have to double check literature but I don't remember seeing this being found in the TCGA database literature anywhere.
```

```{r}
##################
###  COL17A1  ####
##################

df[df$COL17A1 == -2,]$COL17A1 <- 'Deletion'
df[df$COL17A1 == -1,]$COL17A1 <- 'Deletion'
df[df$COL17A1 == 0,]$COL17A1 <- 'Normal'
df[df$COL17A1 == 1,]$COL17A1 <- 'Duplication'
df[df$COL17A1 == 2,]$COL17A1 <- 'Duplication'

survCOL17A1 <- survfit(Surv(Days.to.Death, status) ~ COL17A1, data = df)
print(survCOL17A1)

ggsurvplot(survCOL17A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL17A1 <- survdiff(Surv(Days.to.Death, status) ~ COL17A1, data = df)
print(logrankCOL17A1)
### No findings.

### This is the last of the less than 50 fold expression genes.
```

```{r}
### Start of GREATER THAN 50 FOLD IN EXPRESSION

##################
###  COL1A2  #####
##################

df[df$COL1A2 == -2,]$COL1A2 <- 'Deletion'
df[df$COL1A2 == -1,]$COL1A2 <- 'Deletion'
df[df$COL1A2 == 0,]$COL1A2 <- 'Normal'
df[df$COL1A2 == 1,]$COL1A2 <- 'Duplication'
df[df$COL1A2 == 2,]$COL1A2 <- 'Duplication'

survCOL1A2 <- survfit(Surv(Days.to.Death, status) ~ COL1A2, data = df)
print(survCOL1A2)

ggsurvplot(survCOL1A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL1A2 <- survdiff(Surv(Days.to.Death, status) ~ COL1A2, data = df)
print(logrankCOL1A2)
# Nothing here
```

```{r}
#################
###  COL15A1  ###
#################

df[df$COL15A1 == -2,]$COL15A1 <- 'Deletion'
df[df$COL15A1 == -1,]$COL15A1 <- 'Deletion'
df[df$COL15A1 == 0,]$COL15A1 <- 'Normal'
df[df$COL15A1 == 1,]$COL15A1 <- 'Duplication'
df[df$COL15A1 == 2,]$COL15A1 <- 'Duplication'

survCOL15A1 <- survfit(Surv(Days.to.Death, status) ~ COL15A1, data = df)
print(survCOL15A1)

ggsurvplot(survCOL15A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL15A1 <- survdiff(Surv(Days.to.Death, status) ~ COL15A1, data = df)
print(logrankCOL15A1)
#Nothing here.  Does show a clear deviation from duplication against the others however.  Potentially worth looking at this again with another dataset somewhere if one ever becomes available.
```

```{r}
#################
###  COL21A1  ###
#################

df[df$COL21A1 == -2,]$COL21A1 <- 'Deletion'
df[df$COL21A1 == -1,]$COL21A1 <- 'Deletion'
df[df$COL21A1 == 0,]$COL21A1 <- 'Normal'
df[df$COL21A1 == 1,]$COL21A1 <- 'Duplication'
df[df$COL21A1 == 2,]$COL21A1 <- 'Duplication'

survCOL21A1 <- survfit(Surv(Days.to.Death, status) ~ COL21A1, data = df)
print(survCOL21A1)

ggsurvplot(survCOL21A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL21A1 <- survdiff(Surv(Days.to.Death, status) ~ COL21A1, data = df)
print(logrankCOL21A1)
### Not statistically significant but shows that CNV duplication might show some better outcomes at 2500 days time

### This was last gene at over 50 folds expression.
```

```{r}
### EXTREMELY HIGH EXPRESSION GENE - COL3A1

################
###  COL3A1  ###
################

#df[df$COL3A1 == -2,]$COL3A1 <- 'Deletion'
df[df$COL3A1 <= -1,]$COL3A1 <- 'Deletion'
df[df$COL3A1 == 0,]$COL3A1 <- 'Normal'
df[df$COL3A1 == 1,]$COL3A1 <- 'Duplication'
df[df$COL3A1 == 2,]$COL3A1 <- 'Duplication'

survCOL3A1 <- survfit(Surv(Days.to.Death, status) ~ COL3A1, data = df)
print(survCOL3A1)

ggsurvplot(survCOL3A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL3A1 <- survdiff(Surv(Days.to.Death, status) ~ COL3A1, data = df)
print(logrankCOL3A1)
### Nothing of interest here.
```

```{r}
#################
###  COL22A1  ###
#################

df[df$COL22A1 == -2,]$COL22A1 <- 'Deletion'
df[df$COL22A1 == -1,]$COL22A1 <- 'Deletion'
df[df$COL22A1 == 0,]$COL22A1 <- 'Normal'
df[df$COL22A1 == 1,]$COL22A1 <- 'Duplication'
df[df$COL22A1 == 2,]$COL22A1 <- 'Duplication'

survCOL22A1 <- survfit(Surv(Days.to.Death, status) ~ COL22A1, data = df)
print(survCOL22A1)

ggsurvplot(survCOL22A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL22A1 <- survdiff(Surv(Days.to.Death, status) ~ COL22A1, data = df)
print(logrankCOL22A1)
### This was previously found in etoposide patients in brain cancers to show a decrease in survival between groups.  However, in Ovarian cancer this is not the case.  This collagen gene is also not as commonly expressed in the ovaries as well.
```

```{r}
### LOOKING AT REST OF THE GENES THAT WERE NOT LISTED AT BEGINNING.  

### I will post all statistically significant findings at end if you want to skip to that part.  Cox Proportional Hazard Models for all significant findings will also be at the ned as well.  These are all iterations of the same coding with the rest of the genes(features) in the dataset.
```

```{r}
#################
###  COL2A1  ###
#################

df[df$COL2A1 == -2,]$COL2A1 <- 'Deletion'
df[df$COL2A1 == -1,]$COL2A1 <- 'Deletion'
df[df$COL2A1 == 0,]$COL2A1 <- 'Normal'
df[df$COL2A1 == 1,]$COL2A1 <- 'Duplication'
df[df$COL2A1 == 2,]$COL2A1 <- 'Duplication'

survCOL2A1 <- survfit(Surv(Days.to.Death, status) ~ COL2A1, data = df)
print(survCOL2A1)

ggsurvplot(survCOL2A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL2A1 <- survdiff(Surv(Days.to.Death, status) ~ COL2A1, data = df)
print(logrankCOL2A1)
### Nothing here
```

```{r}
#################
###  COL4A1  ###
#################

df[df$COL4A1 == -2,]$COL4A1 <- 'Deletion'
df[df$COL4A1 == -1,]$COL4A1 <- 'Deletion'
df[df$COL4A1 == 0,]$COL4A1 <- 'Normal'
df[df$COL4A1 == 1,]$COL4A1 <- 'Duplication'
df[df$COL4A1 == 2,]$COL4A1 <- 'Duplication'

survCOL4A1 <- survfit(Surv(Days.to.Death, status) ~ COL4A1, data = df)
print(survCOL4A1)

ggsurvplot(survCOL4A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A1 <- survdiff(Surv(Days.to.Death, status) ~ COL4A1, data = df)
print(logrankCOL4A1)
### Nothing here
```

```{r}
#################
###  COL4A2  ###
#################

df[df$COL4A2 == -2,]$COL4A2 <- 'Deletion'
df[df$COL4A2 == -1,]$COL4A2 <- 'Deletion'
df[df$COL4A2 == 0,]$COL4A2 <- 'Normal'
df[df$COL4A2 == 1,]$COL4A2 <- 'Duplication'
df[df$COL4A2 == 2,]$COL4A2 <- 'Duplication'

survCOL4A2 <- survfit(Surv(Days.to.Death, status) ~ COL4A2, data = df)
print(survCOL4A2)

ggsurvplot(survCOL4A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A2 <- survdiff(Surv(Days.to.Death, status) ~ COL4A2, data = df)
print(logrankCOL4A2)
### Nothing here either
```

```{r}
#################
###  COL4A3  ###
#################

df[df$COL4A3 == -2,]$COL4A3 <- 'Deletion'
df[df$COL4A3 == -1,]$COL4A3 <- 'Deletion'
df[df$COL4A3 == 0,]$COL4A3 <- 'Normal'
df[df$COL4A3 == 1,]$COL4A3 <- 'Duplication'
df[df$COL4A3 == 2,]$COL4A3 <- 'Duplication'

survCOL4A3 <- survfit(Surv(Days.to.Death, status) ~ COL4A3, data = df)
print(survCOL4A3)

ggsurvplot(survCOL4A3,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A3 <- survdiff(Surv(Days.to.Death, status) ~ COL4A3, data = df)
print(logrankCOL4A3)
### Nothing here either
```

```{r}
##################
###  COL4A3BP  ###
##################

df[df$COL4A3BP == 0,]$COL4A3BP <- 'Normal'
df[df$COL4A3BP == -2,]$COL4A3BP <- 'Deletion'
df[df$COL4A3BP == -1,]$COL4A3BP <- 'Deletion'
df[df$COL4A3BP == 1,]$COL4A3BP <- 'Duplication'
df[df$COL4A3BP == 2,]$COL4A3BP <- 'Duplication'

survCOL4A3BP <- survfit(Surv(Days.to.Death, status) ~ COL4A3BP, data = df)
print(survCOL4A3BP)

ggsurvplot(survCOL4A3BP,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A3BP <- survdiff(Surv(Days.to.Death, status) ~ COL4A3BP, data = df)
print(logrankCOL4A3BP)
### Very significant and interesting plot.  The variance in survival times is quite clear here.  This is a binding protein gene seen in collagen.
```

```{r}
################
###  COL4A4  ###
################

df[df$COL4A4 == -2,]$COL4A4 <- 'Deletion'
df[df$COL4A4 == -1,]$COL4A4 <- 'Deletion'
df[df$COL4A4 == 0,]$COL4A4 <- 'Normal'
df[df$COL4A4 == 1,]$COL4A4 <- 'Duplication'
df[df$COL4A4 == 2,]$COL4A4 <- 'Duplication'

survCOL4A4 <- survfit(Surv(Days.to.Death, status) ~ COL4A4, data = df)
print(survCOL4A4)

ggsurvplot(survCOL4A4,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A4 <- survdiff(Surv(Days.to.Death, status) ~ COL4A4, data = df)
print(logrankCOL4A4)
### Nothing here
```

```{r}
################
###  COL4A5  ###
################

df[df$COL4A5 == -2,]$COL4A5 <- 'Deletion'
df[df$COL4A5 == -1,]$COL4A5 <- 'Deletion'
df[df$COL4A5 == 0,]$COL4A5 <- 'Normal'
df[df$COL4A5 == 1,]$COL4A5 <- 'Duplication'
df[df$COL4A5 == 2,]$COL4A5 <- 'Duplication'

survCOL4A5 <- survfit(Surv(Days.to.Death, status) ~ COL4A5, data = df)
print(survCOL4A5)

ggsurvplot(survCOL4A5,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A5 <- survdiff(Surv(Days.to.Death, status) ~ COL4A5, data = df)
print(logrankCOL4A5)
### Nothing here
```

```{r}
################
###  COL4A6  ###
################

df[df$COL4A6 == -2,]$COL4A6 <- 'Deletion'
df[df$COL4A6 == -1,]$COL4A6 <- 'Deletion'
df[df$COL4A6 == 0,]$COL4A6 <- 'Normal'
df[df$COL4A6 == 1,]$COL4A6 <- 'Duplication'
df[df$COL4A6 == 2,]$COL4A6 <- 'Duplication'

survCOL4A6 <- survfit(Surv(Days.to.Death, status) ~ COL4A6, data = df)
print(survCOL4A6)

ggsurvplot(survCOL4A6,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL4A6 <- survdiff(Surv(Days.to.Death, status) ~ COL4A6, data = df)
print(logrankCOL4A6)
### Not much here either
```

```{r}
################
###  COL5A1  ###
################

df[df$COL5A1 == -2,]$COL5A1 <- 'Deletion'
df[df$COL5A1 == -1,]$COL5A1 <- 'Deletion'
df[df$COL5A1 == 0,]$COL5A1 <- 'Normal'
df[df$COL5A1 == 1,]$COL5A1 <- 'Duplication'
df[df$COL5A1 == 2,]$COL5A1 <- 'Duplication'

survCOL5A1 <- survfit(Surv(Days.to.Death, status) ~ COL5A1, data = df)
print(survCOL5A1)

ggsurvplot(survCOL5A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL5A1 <- survdiff(Surv(Days.to.Death, status) ~ COL5A1, data = df)
print(logrankCOL5A1)
### Nothing here
```

```{r}
################
###  COL5A3  ###
################

df[df$COL5A3 == -2,]$COL5A3 <- 'Deletion'
df[df$COL5A3 == -1,]$COL5A3 <- 'Deletion'
df[df$COL5A3 == 0,]$COL5A3 <- 'Normal'
df[df$COL5A3 == 1,]$COL5A3 <- 'Duplication'
df[df$COL5A3 == 2,]$COL5A3 <- 'Duplication'

survCOL5A3 <- survfit(Surv(Days.to.Death, status) ~ COL5A3, data = df)
print(survCOL5A3)

ggsurvplot(survCOL5A3,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL5A3 <- survdiff(Surv(Days.to.Death, status) ~ COL5A3, data = df)
print(logrankCOL5A3)
### Significant.  This gene encodes for low abundance fibrillar collagens.
```

```{r}
################
###  COL6A1  ###
################

df[df$COL6A1  == -2,]$COL6A1  <- 'Deletion'
df[df$COL6A1  == -1,]$COL6A1  <- 'Deletion'
df[df$COL6A1  == 0,]$COL6A1  <- 'Normal'
df[df$COL6A1  == 1,]$COL6A1  <- 'Duplication'
df[df$COL6A1  == 2,]$COL6A1  <- 'Duplication'

survCOL6A1  <- survfit(Surv(Days.to.Death, status) ~ COL6A1 , data = df)
print(survCOL6A1)

ggsurvplot(survCOL6A1 ,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A1 <- survdiff(Surv(Days.to.Death, status) ~ COL6A1, data = df)
print(logrankCOL6A1)
### Nothing here
```

```{r}
################
###  COL6A2  ###
################

df[df$COL6A2  == -2,]$COL6A2  <- 'Deletion'
df[df$COL6A2  == -1,]$COL6A2  <- 'Deletion'
df[df$COL6A2  == 0,]$COL6A2  <- 'Normal'
df[df$COL6A2  == 1,]$COL6A2  <- 'Duplication'
df[df$COL6A2  == 2,]$COL6A2  <- 'Duplication'

survCOL6A2 <- survfit(Surv(Days.to.Death, status) ~ COL6A2 , data = df)
print(survCOL6A2)

ggsurvplot(survCOL6A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A2 <- survdiff(Surv(Days.to.Death, status) ~ COL6A2, data = df)
print(logrankCOL6A2)
### Nothing here
```

```{r}
################
###  COL6A3  ###
################

df[df$COL6A3  == -2,]$COL6A3  <- 'Deletion'
df[df$COL6A3  == -1,]$COL6A3  <- 'Deletion'
df[df$COL6A3  == 0,]$COL6A3  <- 'Normal'
df[df$COL6A3  == 1,]$COL6A3  <- 'Duplication'
df[df$COL6A3  == 2,]$COL6A3  <- 'Duplication'

survCOL6A3 <- survfit(Surv(Days.to.Death, status) ~ COL6A2 , data = df)
print(survCOL6A3)

ggsurvplot(survCOL6A3,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A3 <- survdiff(Surv(Days.to.Death, status) ~ COL6A3, data = df)
print(logrankCOL6A3)
### Nothing here
```

```{r}
##################
###  COL6A4P1  ###
##################

df[df$COL6A4P1  == -2,]$COL6A4P1  <- 'Deletion'
df[df$COL6A4P1  == -1,]$COL6A4P1  <- 'Deletion'
df[df$COL6A4P1  == 0,]$COL6A4P1  <- 'Normal'
df[df$COL6A4P1  == 1,]$COL6A4P1  <- 'Duplication'
df[df$COL6A4P1  == 2,]$COL6A4P1  <- 'Duplication'

survCOL6A4P1 <- survfit(Surv(Days.to.Death, status) ~ COL6A4P1 , data = df)
print(survCOL6A4P1)

ggsurvplot(survCOL6A4P1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A4P1 <- survdiff(Surv(Days.to.Death, status) ~ COL6A4P1, data = df)
print(logrankCOL6A4P1)
### Nothing here
```

```{r}
##################
###  COL6A4P2  ###
##################

#df[df$COL6A4P2  == -2,]$COL6A4P2  <- 'Deletion'
df[df$COL6A4P2  <= -1,]$COL6A4P2  <- 'Deletion'
df[df$COL6A4P2  == 0,]$COL6A4P2  <- 'Normal'
df[df$COL6A4P2  == 1,]$COL6A4P2  <- 'Duplication'
df[df$COL6A4P2  == 2,]$COL6A4P2  <- 'Duplication'

survCOL6A4P2 <- survfit(Surv(Days.to.Death, status) ~ COL6A4P2, data = df)
print(survCOL6A4P2)

ggsurvplot(survCOL6A4P2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A4P2 <- survdiff(Surv(Days.to.Death, status) ~ COL6A4P2, data = df)
print(logrankCOL6A4P2)
### Nothing here
```

```{r}
################
###  COL6A5  ###
################

#df[df$COL6A5  == -2,]$COL6A5  <- 'Deletion'
df[df$COL6A5  <= -1,]$COL6A5  <- 'Deletion'
df[df$COL6A5  == 0,]$COL6A5  <- 'Normal'
df[df$COL6A5  == 1,]$COL6A5  <- 'Duplication'
df[df$COL6A5  == 2,]$COL6A5  <- 'Duplication'

survCOL6A5 <- survfit(Surv(Days.to.Death, status) ~ COL6A5, data = df)
print(survCOL6A5)

ggsurvplot(survCOL6A5,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A5 <- survdiff(Surv(Days.to.Death, status) ~ COL6A5, data = df)
print(logrankCOL6A5)
### Nothing here
```

```{r}
##################
###  COL6A6  ###
##################

#df[df$COL6A6  == -2,]$COL6A6  <- 'Deletion'
df[df$COL6A6  <= -1,]$COL6A6  <- 'Deletion'
df[df$COL6A6  == 0,]$COL6A6  <- 'Normal'
df[df$COL6A6  == 1,]$COL6A6  <- 'Duplication'
df[df$COL6A6  == 2,]$COL6A6  <- 'Duplication'

survCOL6A6 <- survfit(Surv(Days.to.Death, status) ~ COL6A6, data = df)
print(survCOL6A5)

ggsurvplot(survCOL6A6,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL6A6 <- survdiff(Surv(Days.to.Death, status) ~ COL6A6, data = df)
print(logrankCOL6A6)
### Nothing here
```

```{r}
################
###  COL7A1  ###
################

df[df$COL7A1  == -2,]$COL7A1  <- 'Deletion'
df[df$COL7A1  == -1,]$COL7A1  <- 'Deletion'
df[df$COL7A1  == 0,]$COL7A1  <- 'Normal'
df[df$COL7A1  == 1,]$COL7A1  <- 'Duplication'
df[df$COL7A1  == 2,]$COL7A1  <- 'Duplication'

survCOL7A1 <- survfit(Surv(Days.to.Death, status) ~ COL7A1, data = df)
print(survCOL7A1)

ggsurvplot(survCOL7A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL7A1 <- survdiff(Surv(Days.to.Death, status) ~ COL7A1, data = df)
print(logrankCOL7A1)
### Nothing here
```

```{r}
################
###  COL8A1  ###
################

#df[df$COL8A1  == -2,]$COL8A1  <- 'Deletion'
df[df$COL8A1  <= -1,]$COL8A1  <- 'Deletion'
df[df$COL8A1  == 0,]$COL8A1  <- 'Normal'
df[df$COL8A1  == 1,]$COL8A1  <- 'Duplication'
df[df$COL8A1  == 2,]$COL8A1  <- 'Duplication'

survCOL8A1 <- survfit(Surv(Days.to.Death, status) ~ COL8A1, data = df)
print(survCOL8A1)

ggsurvplot(survCOL8A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL8A1 <- survdiff(Surv(Days.to.Death, status) ~ COL8A1, data = df)
print(logrankCOL8A1)
### Nothing here
```

```{r}
################
###  COL8A2  ###
################

df[df$COL8A2  == -2,]$COL8A2  <- 'Deletion'
df[df$COL8A2  == -1,]$COL8A2  <- 'Deletion'
df[df$COL8A2  == 0,]$COL8A2  <- 'Normal'
df[df$COL8A2  == 1,]$COL8A2  <- 'Duplication'
df[df$COL8A2  == 2,]$COL8A2  <- 'Duplication'

survCOL8A2 <- survfit(Surv(Days.to.Death, status) ~ COL8A2, data = df)
print(survCOL8A2)

ggsurvplot(survCOL8A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL8A2 <- survdiff(Surv(Days.to.Death, status) ~ COL8A2, data = df)
print(logrankCOL8A2)
### Nothing here
```

```{r}
################
###  COL9A1  ###
################

#df[df$COL9A1  == -2,]$COL9A1  <- 'Deletion'
df[df$COL9A1  <= -1,]$COL9A1 <- 'Deletion'
df[df$COL9A1  == 0,]$COL9A1  <- 'Normal'
df[df$COL9A1  == 1,]$COL9A1  <- 'Duplication'
df[df$COL9A1  == 2,]$COL9A1  <- 'Duplication'

survCOL9A1 <- survfit(Surv(Days.to.Death, status) ~ COL9A1, data = df)
print(survCOL9A1)

ggsurvplot(survCOL9A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL9A1 <- survdiff(Surv(Days.to.Death, status) ~ COL9A1, data = df)
print(logrankCOL9A1)
### Nothing here
```

```{r}
################
###  COL9A2  ###
################

df[df$COL9A2  == -2,]$COL9A2  <- 'Deletion'
df[df$COL9A2  == -1,]$COL9A2 <- 'Deletion'
df[df$COL9A2  == 0,]$COL9A2  <- 'Normal'
df[df$COL9A2  == 1,]$COL9A2  <- 'Duplication'
df[df$COL9A2  == 2,]$COL9A2  <- 'Duplication'

survCOL9A2 <- survfit(Surv(Days.to.Death, status) ~ COL9A2, data = df)
print(survCOL9A2)

ggsurvplot(survCOL9A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL9A2 <- survdiff(Surv(Days.to.Death, status) ~ COL9A2, data = df)
print(logrankCOL9A2)
### Nothing here
```

```{r}
################
###  COL9A3  ###
################

#df[df$COL9A3  == -2,]$COL9A3  <- 'Deletion'
df[df$COL9A3  <= -1,]$COL9A3 <- 'Deletion'
df[df$COL9A3  == 0,]$COL9A3  <- 'Normal'
df[df$COL9A3  == 1,]$COL9A3  <- 'Duplication'
df[df$COL9A3  == 2,]$COL9A3  <- 'Duplication'

survCOL9A3 <- survfit(Surv(Days.to.Death, status) ~ COL9A3, data = df)
print(survCOL9A3)

ggsurvplot(survCOL9A3,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL9A3 <- survdiff(Surv(Days.to.Death, status) ~ COL9A3, data = df)
print(logrankCOL9A3)
### Nothing here
```

```{r}
#################
###  COL10A1  ###
#################

df[df$COL10A1  == -2,]$COL10A1  <- 'Deletion'
df[df$COL10A1  == -1,]$COL10A1 <- 'Deletion'
df[df$COL10A1  == 0,]$COL10A1  <- 'Normal'
df[df$COL10A1  == 1,]$COL10A1  <- 'Duplication'
df[df$COL10A1  == 2,]$COL10A1  <- 'Duplication'

survCOL10A1 <- survfit(Surv(Days.to.Death, status) ~ COL10A1, data = df)
print(survCOL10A1)

ggsurvplot(survCOL10A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL10A1 <- survdiff(Surv(Days.to.Death, status) ~ COL10A1, data = df)
print(logrankCOL10A1)
### Nothing here
```

```{r}
#################
###  COL11A1  ###
#################

df[df$COL11A1  == -2,]$COL11A1  <- 'Deletion'
df[df$COL11A1  == -1,]$COL11A1 <- 'Deletion'
df[df$COL11A1  == 0,]$COL11A1  <- 'Normal'
df[df$COL11A1  == 1,]$COL11A1  <- 'Duplication'
df[df$COL11A1  == 2,]$COL11A1  <- 'Duplication'

survCOL11A1 <- survfit(Surv(Days.to.Death, status) ~ COL11A1, data = df)
print(survCOL11A1)

ggsurvplot(survCOL11A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL11A1 <- survdiff(Surv(Days.to.Death, status) ~ COL11A1, data = df)
print(logrankCOL11A1)
### Nothing here
```

```{r}
#################
###  COL11A2  ###
#################

#df[df$COL11A2  == -2,]$COL11A2  <- 'Deletion'
df[df$COL11A2  <= -1,]$COL11A2 <- 'Deletion'
df[df$COL11A2  == 0,]$COL11A2  <- 'Normal'
df[df$COL11A2  == 1,]$COL11A2  <- 'Duplication'
df[df$COL11A2  == 2,]$COL11A2  <- 'Duplication'

survCOL11A2 <- survfit(Surv(Days.to.Death, status) ~ COL11A2, data = df)
print(survCOL11A1)

ggsurvplot(survCOL11A2,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL11A2 <- survdiff(Surv(Days.to.Death, status) ~ COL11A2, data = df)
print(logrankCOL11A2)
### Nothing here
```

```{r}
#################
###  COL13A1  ###
#################

df[df$COL13A1  == -2,]$COL13A1  <- 'Deletion'
df[df$COL13A1  == -1,]$COL13A1 <- 'Deletion'
df[df$COL13A1  == 0,]$COL13A1  <- 'Normal'
df[df$COL13A1  == 1,]$COL13A1  <- 'Duplication'
df[df$COL13A1  == 2,]$COL13A1  <- 'Duplication'

survCOL13A1 <- survfit(Surv(Days.to.Death, status) ~ COL13A1, data = df)
print(survCOL13A1)

ggsurvplot(survCOL13A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL13A1 <- survdiff(Surv(Days.to.Death, status) ~ COL13A1, data = df)
print(logrankCOL13A1)
### Nothing here
```

```{r}
#################
###  COL14A1  ###
#################

df[df$COL14A1  == -2,]$COL14A1  <- 'Deletion'
df[df$COL14A1  == -1,]$COL14A1 <- 'Deletion'
df[df$COL14A1  == 0,]$COL14A1  <- 'Normal'
df[df$COL14A1  == 1,]$COL14A1  <- 'Duplication'
df[df$COL14A1  == 2,]$COL14A1  <- 'Duplication'

survCOL14A1 <- survfit(Surv(Days.to.Death, status) ~ COL14A1, data = df)
print(survCOL14A1)

ggsurvplot(survCOL14A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL14A1 <- survdiff(Surv(Days.to.Death, status) ~ COL14A1, data = df)
print(logrankCOL14A1)
### Nothing here
```

```{r}
#################
###  COL16A1  ###
#################

df[df$COL16A1  == -2,]$COL16A1  <- 'Deletion'
df[df$COL16A1  == -1,]$COL16A1 <- 'Deletion'
df[df$COL16A1  == 0,]$COL16A1  <- 'Normal'
df[df$COL16A1  == 1,]$COL16A1  <- 'Duplication'
df[df$COL16A1  == 2,]$COL16A1  <- 'Duplication'

survCOL16A1 <- survfit(Surv(Days.to.Death, status) ~ COL16A1, data = df)
print(survCOL16A1)

ggsurvplot(survCOL16A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL16A1 <- survdiff(Surv(Days.to.Death, status) ~ COL16A1, data = df)
print(logrankCOL16A1)
### Nothing here
```

```{r}
#################
###  COL18A1  ###
#################

df[df$COL18A1  == -2,]$COL18A1  <- 'Deletion'
df[df$COL18A1  == -1,]$COL18A1 <- 'Deletion'
df[df$COL18A1  == 0,]$COL18A1  <- 'Normal'
df[df$COL18A1  == 1,]$COL18A1  <- 'Duplication'
df[df$COL18A1  == 2,]$COL18A1  <- 'Duplication'

survCOL18A1 <- survfit(Surv(Days.to.Death, status) ~ COL18A1, data = df)
print(survCOL18A1)

ggsurvplot(survCOL18A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL18A1 <- survdiff(Surv(Days.to.Death, status) ~ COL18A1, data = df)
print(logrankCOL18A1)
### Nothing here
```

```{r}
#################
###  COL19A1  ###
#################

#df[df$COL19A1  == -2,]$COL19A1  <- 'Deletion'
df[df$COL19A1  <= -1,]$COL19A1 <- 'Deletion'
df[df$COL19A1  == 0,]$COL19A1  <- 'Normal'
df[df$COL19A1  == 1,]$COL19A1  <- 'Duplication'
df[df$COL19A1  == 2,]$COL19A1  <- 'Duplication'

survCOL19A1 <- survfit(Surv(Days.to.Death, status) ~ COL19A1, data = df)
print(survCOL19A1)

ggsurvplot(survCOL19A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL19A1 <- survdiff(Surv(Days.to.Death, status) ~ COL19A1, data = df)
print(logrankCOL19A1)
### Nothing here
```

```{r}
#################
###  COL20A1  ###
#################

#df[df$COL20A1  == -2,]$COL20A1  <- 'Deletion'
df[df$COL20A1  <= -1,]$COL20A1 <- 'Deletion'
df[df$COL20A1  == 0,]$COL20A1  <- 'Normal'
df[df$COL20A1  == 1,]$COL20A1  <- 'Duplication'
df[df$COL20A1  == 2,]$COL20A1  <- 'Duplication'

survCOL20A1 <- survfit(Surv(Days.to.Death, status) ~ COL20A1, data = df)
print(survCOL20A1)

ggsurvplot(survCOL20A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL20A1 <- survdiff(Surv(Days.to.Death, status) ~ COL20A1, data = df)
print(logrankCOL20A1)
### Nothing here
```

```{r}
#################
###  COL23A1  ###
#################

df[df$COL23A1  == -2,]$COL23A1  <- 'Deletion'
df[df$COL23A1  == -1,]$COL23A1 <- 'Deletion'
df[df$COL23A1  == 0,]$COL23A1  <- 'Normal'
df[df$COL23A1  == 1,]$COL23A1  <- 'Duplication'
df[df$COL23A1  == 2,]$COL23A1  <- 'Duplication'

survCOL23A1 <- survfit(Surv(Days.to.Death, status) ~ COL23A1, data = df)
print(survCOL23A1)

ggsurvplot(survCOL23A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL23A1 <- survdiff(Surv(Days.to.Death, status) ~ COL23A1, data = df)
print(logrankCOL23A1)
### Nothing here
```

```{r}
#################
###  COL24A1  ###
#################

df[df$COL24A1  == -2,]$COL24A1  <- 'Deletion'
df[df$COL24A1  == -1,]$COL24A1 <- 'Deletion'
df[df$COL24A1  == 0,]$COL24A1  <- 'Normal'
df[df$COL24A1  == 1,]$COL24A1  <- 'Duplication'
df[df$COL24A1  == 2,]$COL24A1  <- 'Duplication'

survCOL24A1 <- survfit(Surv(Days.to.Death, status) ~ COL24A1, data = df)
print(survCOL24A1)

ggsurvplot(survCOL24A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL24A1 <- survdiff(Surv(Days.to.Death, status) ~ COL24A1, data = df)
print(logrankCOL24A1)
### Nothing here
```

```{r}
#################
###  COL25A1  ###
#################

df[df$COL25A1  == -2,]$COL25A1  <- 'Deletion'
df[df$COL25A1  == -1,]$COL25A1 <- 'Deletion'
df[df$COL25A1  == 0,]$COL25A1  <- 'Normal'
df[df$COL25A1  == 1,]$COL25A1  <- 'Duplication'
df[df$COL25A1  == 2,]$COL25A1  <- 'Duplication'

survCOL25A1 <- survfit(Surv(Days.to.Death, status) ~ COL25A1, data = df)
print(survCOL25A1)

ggsurvplot(survCOL25A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL25A1 <- survdiff(Surv(Days.to.Death, status) ~ COL25A1, data = df)
print(logrankCOL25A1)
### Nothing here
```

```{r}
#################
###  COL26A1  ###
#################

#df[df$COL26A1  == -2,]$COL26A1  <- 'Deletion'
df[df$COL26A1  <= -1,]$COL26A1 <- 'Deletion'
df[df$COL26A1  == 0,]$COL26A1  <- 'Normal'
df[df$COL26A1  == 1,]$COL26A1  <- 'Duplication'
df[df$COL26A1  == 2,]$COL26A1  <- 'Duplication'

survCOL26A1 <- survfit(Surv(Days.to.Death, status) ~ COL26A1, data = df)
print(survCOL26A1)

ggsurvplot(survCOL26A1,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

logrankCOL26A1 <- survdiff(Surv(Days.to.Death, status) ~ COL26A1, data = df)
print(logrankCOL26A1)
### Nothing here
```

```{r}
############################################################################
### Cox Proportional Hazard Models ###### Cox Proportional Hazard Models ###
############################################################################
############################################################################
### Cox Proportional Hazard Models ###### Cox Proportional Hazard Models ###
############################################################################
############################################################################
### Cox Proportional Hazard Models ###### Cox Proportional Hazard Models ###
############################################################################
############################################################################
### Cox Proportional Hazard Models ###### Cox Proportional Hazard Models ###
############################################################################
############################################################################
### Cox Proportional Hazard Models ###### Cox Proportional Hazard Models ###
############################################################################
```

```{r}
### List of genes that met statistical significance.
### COL12A1, COL4A3BP, COL5A3
attach(df)

#####################
### COL12A1 Model ###
#####################

ggsurvplot(survCOL12A1,
          pval = TRUE, conf.int = FALSE,
          size = 1,
          risk.table = TRUE,
          risk.table.col = "strata",
          fontsize = 3.5,
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

coxCOL12A1 <- coxph(Surv(Days.to.Death, status) ~ COL12A1)
summary(coxCOL12A1)
cox.zph(coxCOL12A1)
plot(cox.zph(coxCOL12A1)[1])
abline(h=0, col=3)

logrankCOL12A1 <- survdiff(Surv(Days.to.Death, status) ~ COL12A1, data = df)
print(logrankCOL12A1)
print(survCOL12A1)

### Looking at second set of data with COL12A1Duplication exp(coef) = 0.6977 we can say that having a duplicated CNV count is 0.70 times better chances of survival or 0.7 to 1 chance of death.  This is the hazard ratio.  This equates to a 30% chance less likely to die with a duplicated CNV of some kind.  The exp(-coef) = 1.433 means that someone without a duplication has a 43% increased chance of dying.

### There was a 52% concordance level.  This probably isn't great fit but it is above 50%.  The graphs is a little noisy when you look at it so it could be expected to have a lower concordance value.

### Looking at the hazard ratio over time, we can see that the beta line doesn't stay within the 95% confidence intervals 100% of the time but barely skirts out at some point.  Our assumptions are mostly met with this model. Can see each stratified group for the gene in the cox hazard plot over time.
```

```{r}
### Cleaner view of cox survival model
CoxsurvCOL12A1 <- survfit(coxCOL12A1)
CoxsurvCOL12A1
summary(CoxsurvCOL12A1, times = 8)
plot(CoxsurvCOL12A1, col = c("black", "red", "blue"),xlab = "Days", 
     ylab = "Overall survival probability")
```

```{r}
### I won't run this on each gene since it is so long but wanted to show the survival differences in numbers instead of the chart.

### Looking at time 2000 days roughly for each one.  Deletion = 19.38% chance of survival.  Duplication = 36.9% Chance of survival.  Normal = 27.81% chance of survival.  These are also clearly seen on the survival plot above.  I picked 2000 days since it was less noisy.
summary(survCOL12A1)
```

```{r}
######################
### COL4A3BP Model ###
######################
ggsurvplot(survCOL4A3BP,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

coxCOL4A3BP <- coxph(Surv(Days.to.Death, status) ~ COL4A3BP)
summary(coxCOL4A3BP)
cox.zph(coxCOL4A3BP)
plot(cox.zph(coxCOL4A3BP)[1])
abline(h=0, col=3)

logrankCOL4A3BP <- survdiff(Surv(Days.to.Death, status) ~ COL4A3BP, data = df)
print(logrankCOL4A3BP)
print(survCOL4A3BP)

### Looking at second set of data with COL4A3BPDuplication exp(coef)= 2.23926 we can say that having a duplicated CNV count is 2.24 times worse chances of survival.  This is the hazard ratio.  So, this is a 124% extra chance of dying if you have this gene duplicated.  The exp(-coef) = 0.4466  means that someone without a duplication has a 45% increased chance of survival.

### There was a 52% concordance level here also.  This probably isn't great fit but it is above 50%.  This grapgh isn't as noisy with regards to duplication and can clearly see the survival issues with this gene.  There is more noise however between deletion and normal CNV count.

### The beta lines seem to stay within the 95% confidence intervals most of the time on this plot of hazard ratio over time, except the very beginning.  Our assumptions are mostly met with this model.  We also can see each strafied group in the cox hazard over time model for this gene.
```

```{r}
### Cleaner look at cox survival plot
CoxsurvCOL4A3BP <- survfit(coxCOL4A3BP)
CoxsurvCOL4A3BP
summary(CoxsurvCOL4A3BP, times = 8)
plot(CoxsurvCOL4A3BP, col = c("black", "red", "blue"))
```

```{r}
####################
### COL5A3 Model ###
####################
ggsurvplot(survCOL5A3,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,
          risk.table.col = "strata",
          linetype = "strata",
          surv.median.line = "hv",
          ggtheme = theme_bw())

coxCOL5A3 <- coxph(Surv(Days.to.Death, status) ~ COL5A3)
summary(coxCOL5A3)
cox.zph(coxCOL5A3)
plot(cox.zph(coxCOL5A3)[1])
abline(h=0, col=3)

logrankCOL5A3 <- survdiff(Surv(Days.to.Death, status) ~ COL5A3, data = df)
print(logrankCOL5A3)
print(survCOL5A3)

### Looking at third set of data with COL5A3Duplication exp(coef)= 0.6827  we can say that having a duplicated CNV count is 0.68 times better chances of survival.  This is the hazard ratio.  So, this is a 32% less chance of dying if you have this gene duplicated.  The exp(-coef) = 1.465  means that someone without a duplication has a 46.5% increased chance of dying.  Also, someone without a duplication or normal CNV count has a 35% increased chance of dying from exp(-coef) = 1.349.

### There was a 55% concordance level here also.  This is the best fit so far of the 3 models.  This grapgh isn't as noisy when it comes to deletions and can clearly see some survival issues with this gene CNV in deletion.  There is more noise however between deuplication and normal CNV count.

### The beta lines do not stay within the 95% confidence intervals at all times on this plot of hazard ratio over time.

### We can also see a clear distinction in all the stratified CNV counts for this gene.
```

```{r}
### Cleaner look at cox survival plot
CoxsurvCOL5A3 <- survfit(coxCOL5A3)
CoxsurvCOL5A3
summary(CoxsurvCOL5A3, times=8)
plot(CoxsurvCOL5A3, col = c("black", "red", "blue"))

```

```{r}
summary(coxCOL5A3)

```


```{r}
##################
### References ###
##################

### Buckley, A. R., Standish, K. A., Bhutani, K., Ideker, T., Lasken, R. S., Carter, H., Harismendy, O., & Schork, N. J. (2017). Pan-cancer analysis reveals technical artifacts in TCGA germline variant calls. BMC Genomics, 18(1). https://doi.org/10.1186/s12864-017-3770-y

### Cheon, D.-J., Tong, Y., Sim, M.-S., Dering, J., Berel, D., Cui, X., Lester, J., Beach, J. A., Tighiouart, M., Walts, A. E., Karlan, B. Y., & Orsulic, S. (2013). A Collagen-Remodeling Gene Signature Regulated by TGF-β Signaling Is Associate=d with Metastasis and Poor Survival in Serous Ovarian Cancer. Clinical Cancer Research, 20(3), 711–723. https://doi.org/10.1158/1078-0432.ccr-13-1256

### Dillies, M.-A., Rau, A., Aubert, J., Hennequet-Antier, C., Jeanmougin, M., Servant, N., Keime, C., Marot, G., Castel, D., Estelle, J., Guernec, G., Jagla, B., Jouneau, L., Laloe, D., Le Gall, C., Schaeffer, B., Le Crom, S., Guedj, M., & Jaffrezic, F. (2012). A comprehensive evaluation of normalization methods for Illumina high-throughput RNA sequencing data analysis. Briefings in Bioinformatics, 14(6), 671–683. https://doi.org/10.1093/bib/bbs046

### Ganapathi, M. K., Jones, W. D., Sehouli, J., Michener, C. M., Braicu, I. E., Norris, E. J., Biscotti, C. V., Vaziri, S. A. J., & Ganapathi, R. N. (2015). Expression profile of COL2A1 and the pseudogene SLC6A10P predicts tumor recurrence in high-grade serous ovarian cancer. International Journal of Cancer, 138(3), 679–688. https://doi.org/10.1002/ijc.29815

### Januchowski, R., Świerczewska, M., Sterzyńska, K., Wojtowicz, K., Nowicki, M., & Zabel, M. (2016). Increased Expression of Several Collagen Genes is Associated with Drug Resistance in Ovarian Cancer Cell Lines. Journal of Cancer, 7(10), 1295–1310. https://doi.org/10.7150/jca.15371

### Liu, J., Lichtenberg, T., Hoadley, K. A., Poisson, L. M., Lazar, A. J., Cherniack, A. D., Kovatich, A. J., Benz, C. C., Levine, D. A., Lee, A. V., Omberg, L., Wolf, D. M., Shriver, C. D., Thorsson, V., Hu, H., Caesar-Johnson, S. J., Demchok, J. A., Felau, I., Kasapi, M., … Mariamidze, A. (2018). An Integrated TCGA Pan-Cancer Clinical Data Resource to Drive High-Quality Survival Outcome Analytics. Cell, 173(2), 400-416.e11. https://doi.org/10.1016/j.cell.2018.02.052

### Package “survminer” Type Package Title Drawing Survival Curves using “ggplot2.” (2019). https://cran.r-project.org/web/packages/survminer/survminer.pdf

### Ricciardelli, C., & Rodgers, R. (2006). Extracellular Matrix of Ovarian Tumors. Seminars in Reproductive Medicine, 24(4), 270–282. https://doi.org/10.1055/s-2006-948556

### Spainhour, J. C. G., & Qiu, P. (2016). Identification of gene-drug interactions that impact patient survival in TCGA. BMC Bioinformatics, 17(1). https://doi.org/10.1186/s12859-016-1255-7

### Teng, P.-N., Wang, G., Hood, B. L., Conrads, K. A., Hamilton, C. A., Maxwell, G. L., Darcy, K. M., & Conrads, T. P. (2013). Identification of candidate circulating cisplatin-resistant biomarkers from epithelial ovarian carcinoma cell secretomes. British Journal of Cancer, 110(1), 123–132. https://doi.org/10.1038/bjc.2013.687

### Wu, Y.-H., Chang, T.-H., Huang, Y.-F., Huang, H.-D., & Chou, C.-Y. (2013). COL11A1 promotes tumor progression and predicts poor clinical outcome in ovarian cancer. Oncogene, 33(26), 3432–3440. https://doi.org/10.1038/onc.2013.307

### Xu, S., Xu, H., Wang, W., Li, S., Li, H., Li, T., Zhang, W., Yu, X., & Liu, L. (2019). The role of collagen in cancer: from bench to bedside. Journal of Translational Medicine, 17(1). https://doi.org/10.1186/s12967-019-2058-1
```